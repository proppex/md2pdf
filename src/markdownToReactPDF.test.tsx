import markdownToReactPDF from "./markdownToReactPDF";

describe("markdownToReactPDF", () => {
  it("should render a root node with children", () => {
    const markdown = `
      # Heading 1
      ## Heading 2
      ### Heading 3
      Paragraph
    `;
    const result = markdownToReactPDF(markdown);
    expect(result).toBeDefined();
  });

  // it("should render a heading node with style", () => {
  //   const markdown = `
  //     # Heading 1
  //     ## Heading 2
  //     ### Heading 3
  //   `;
  //   const result = markdownToReactPDF(markdown);
  //   const container = render(result);
  //   expect(container).toMatchSnapshot();
  // });

  // it("should render a paragraph node with style", () => {
  //   const markdown = `
  //     Paragraph
  //   `;
  //   const result = markdownToReactPDF(markdown);
  //   const container = render(result);
  //   expect(container).toMatchSnapshot();
  // });

  // it("should render an emphasis node with style", () => {
  //   const markdown = `
  //     *Emphasis*
  //     **Strong**
  //   `;
  //   const result = markdownToReactPDF(markdown);
  //   const container = render(result);
  //   expect(container).toMatchSnapshot();
  // });

  // it("should render an image node with style", () => {
  //   const markdown = `
  //     ![Company Logo](https://example.com/logo.png)
  //   `;
  //   const result = markdownToReactPDF(markdown);
  //   const container = render(result);
  //   expect(container).toMatchSnapshot();
  // });
});
