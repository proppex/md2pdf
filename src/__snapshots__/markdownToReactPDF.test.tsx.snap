// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`markdownToReactPDF > should render a heading node with style 1`] = `
<VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontSize": 20,
          "marginBottom": 15,
        }
      }
    >
      Heading 1
    </TEXT>
  </VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontSize": 16,
          "marginBottom": 12,
        }
      }
    >
      Heading 2
    </TEXT>
  </VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontSize": 14,
          "marginBottom": 10,
        }
      }
    >
      Heading 3
    </TEXT>
  </VIEW>
</VIEW>
`;

exports[`markdownToReactPDF > should render a paragraph node with style 1`] = `
<VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontFamily": "Times-Roman",
          "fontSize": 12,
          "marginBottom": 10,
        }
      }
    >
      <TEXT>
        Paragraph
      </TEXT>
    </TEXT>
  </VIEW>
</VIEW>
`;

exports[`markdownToReactPDF > should render a root node with children 1`] = `
<VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontSize": 20,
          "marginBottom": 15,
        }
      }
    >
      Heading 1
    </TEXT>
  </VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontSize": 16,
          "marginBottom": 12,
        }
      }
    >
      Heading 2
    </TEXT>
  </VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontSize": 14,
          "marginBottom": 10,
        }
      }
    >
      Heading 3
    </TEXT>
  </VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontFamily": "Times-Roman",
          "fontSize": 12,
          "marginBottom": 10,
        }
      }
    >
      <TEXT>
        Paragraph
      </TEXT>
    </TEXT>
  </VIEW>
</VIEW>
`;

exports[`markdownToReactPDF > should render an emphasis node with style 1`] = `
<VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontFamily": "Times-Roman",
          "fontSize": 12,
          "marginBottom": 10,
        }
      }
    >
      <TEXT>
        <TEXT
          style={
            {
              "fontSize": 12,
              "textDecoration": "underline",
            }
          }
        >
          Emphasis
        </TEXT>
      </TEXT>
      <TEXT>
        

      </TEXT>
    </TEXT>
  </VIEW>
</VIEW>
`;

exports[`markdownToReactPDF > should render an image node with style 1`] = `
<VIEW>
  <VIEW>
    <TEXT
      style={
        {
          "fontFamily": "Times-Roman",
          "fontSize": 12,
          "marginBottom": 10,
        }
      }
    >
      <IMAGE
        src="https://example.com/logo.png"
        style={
          {
            "height": 23,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 20,
            "width": 23,
          }
        }
      />
    </TEXT>
  </VIEW>
</VIEW>
`;
